In [12]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as stats

# Loading the Data

In [24]:
df_2000 = pd.read_csv('/Users/christianrim/Documents/final_tmdb_data_2000.csv',lineterminator='\n')
df_2001 = pd.read_csv('/Users/christianrim/Documents/final_tmdb_data_2001.csv',lineterminator='\n')

In [25]:
# display the 2000 movie dataset
df_2000.head()
df_2001.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6153 entries, 0 to 6152
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                6153 non-null   object 
 1   adult                  6152 non-null   float64
 2   backdrop_path          1782 non-null   object 
 3   belongs_to_collection  265 non-null    object 
 4   budget                 6152 non-null   float64
 5   genres                 6152 non-null   object 
 6   homepage               391 non-null    object 
 7   id                     6152 non-null   float64
 8   original_language      6152 non-null   object 
 9   original_title         6152 non-null   object 
 10  overview               5409 non-null   object 
 11  popularity             6152 non-null   float64
 12  poster_path            4591 non-null   object 
 13  production_companies   6152 non-null   object 
 14  production_countries   6152 non-null   object 
 15  rele

In [26]:
df =  pd.concat([df_2000,df_2001])
df.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0015414    0.0                               NaN                   NaN   
2  tt0113026    0.0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg                   NaN   
3  tt0113086    0.0                               NaN                   NaN   
4  tt0113092    0.0                               NaN                   NaN   

       budget                                             genres homepage  \
0         NaN                                                NaN      NaN   
1         0.0                      [{'id': 18, 'name': 'Drama'}]      NaN   
2  10000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...      NaN   
3         0.0                                                 []      NaN   
4         0.0           [{'id': 878, 'name': 'Science Fiction'}]      NaN   

         id original_language          original_title  ... revenue  runtime  \
0       NaN               NaN                     NaN  ...     NaN      NaN   
1  607290.0                es  La tierra de los toros  ...     0.0     53.0   
2   62127.0                en         The Fantasticks  ...     0.0     86.0   
3  612666.0                en  Florentino y el diablo  ...     0.0     90.0   
4  110977.0                en           For the Cause  ...     0.0    100.0   

                                    spoken_languages    status  \
0                                                NaN       NaN   
1  [{'english_name': 'No Language', 'iso_639_1': ...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3                                                 []  Released   
4  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                                         tagline                   title  \
0                                            NaN                     NaN   
1                                            NaN   The Land of the Bulls   
2  Try to remember the first time magic happened         The Fantasticks   
3                                            NaN  Florentino y el diablo   
4   The ultimate showdown on a forbidden planet.           For the Cause   

   video  vote_average vote_count certification  
0    NaN           NaN        NaN           NaN  
1    0.0           0.0        0.0           NaN  
2    0.0           5.5       22.0           NaN  
3    0.0           5.0        1.0           NaN  
4    0.0           5.1        8.0           NaN  

[5 rows x 26 columns]

In [27]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 11978 entries, 0 to 6152
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                11978 non-null  object 
 1   adult                  11976 non-null  float64
 2   backdrop_path          3456 non-null   object 
 3   belongs_to_collection  519 non-null    object 
 4   budget                 11976 non-null  float64
 5   genres                 11976 non-null  object 
 6   homepage               701 non-null    object 
 7   id                     11976 non-null  float64
 8   original_language      11976 non-null  object 
 9   original_title         11976 non-null  object 
 10  overview               10467 non-null  object 
 11  popularity             11976 non-null  float64
 12  poster_path            8972 non-null   object 
 13  production_companies   11976 non-null  object 
 14  production_countries   11976 non-null  object 
 15  rel

# Cleaning the data

In [28]:
# Removing columns with a lot of missing values
df = df.drop(['belongs_to_collection','backdrop_path','poster_path','tagline','homepage'], axis=1)
df.head()

imdb_id  adult      budget  \
0          0    NaN         NaN   
1  tt0015414    0.0         0.0   
2  tt0113026    0.0  10000000.0   
3  tt0113086    0.0         0.0   
4  tt0113092    0.0         0.0   

                                              genres        id  \
0                                                NaN       NaN   
1                      [{'id': 18, 'name': 'Drama'}]  607290.0   
2  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...   62127.0   
3                                                 []  612666.0   
4           [{'id': 878, 'name': 'Science Fiction'}]  110977.0   

  original_language          original_title  \
0               NaN                     NaN   
1                es  La tierra de los toros   
2                en         The Fantasticks   
3                en  Florentino y el diablo   
4                en           For the Cause   

                                            overview  popularity  \
0                                                NaN         NaN   
1                                                NaN       0.600   
2  Two rural teens sing and dance their way throu...       3.081   
3  Florentino is a young horseman and singer, a f...       0.600   
4  Earth is in a state of constant war and two co...       2.099   

                                production_companies  ... release_date  \
0                                                NaN  ...          NaN   
1  [{'id': 77948, 'logo_path': None, 'name': 'Mus...  ...   1924-01-01   
2  [{'id': 51207, 'logo_path': None, 'name': 'Sul...  ...   2000-09-22   
3                                                 []  ...   2000-04-29   
4  [{'id': 7405, 'logo_path': '/rfnws0uY8rsNAsrLb...  ...   2000-11-15   

  revenue  runtime                                   spoken_languages  \
0     NaN      NaN                                                NaN   
1     0.0     53.0  [{'english_name': 'No Language', 'iso_639_1': ...   
2     0.0     86.0  [{'english_name': 'English', 'iso_639_1': 'en'...   
3     0.0     90.0                                                 []   
4     0.0    100.0  [{'english_name': 'English', 'iso_639_1': 'en'...   

     status                   title video  vote_average  vote_count  \
0       NaN                     NaN   NaN           NaN         NaN   
1  Released   The Land of the Bulls   0.0           0.0         0.0   
2  Released         The Fantasticks   0.0           5.5        22.0   
3  Released  Florentino y el diablo   0.0           5.0         1.0   
4  Released           For the Cause   0.0           5.1         8.0   

   certification  
0            NaN  
1            NaN  
2            NaN  
3            NaN  
4            NaN  

[5 rows x 21 columns]

In [32]:
df['budget'].describe()

count    1.197600e+04
mean     1.146149e+06
std      8.177762e+06
min      0.000000e+00
25%      0.000000e+00
50%      0.000000e+00
75%      0.000000e+00
max      1.400000e+08
Name: budget, dtype: float64

In [33]:
# getting rid of the 0's in the 'budget' and 'revenue' column
filtered_df = df[(df['budget'] != 0) & (df['revenue'] != 0)]

In [34]:
filtered_df.head()

imdb_id  adult      budget  \
0           0    NaN         NaN   
7   tt0118694    0.0    150000.0   
19  tt0120467    0.0    120000.0   
22  tt0120630    0.0  45000000.0   
25  tt0120753    0.0   8000000.0   

                                               genres       id  \
0                                                 NaN      NaN   
7   [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...    843.0   
19  [{'id': 18, 'name': 'Drama'}, {'id': 27, 'name...  19085.0   
22  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   7443.0   
25  [{'id': 18, 'name': 'Drama'}, {'id': 53, 'name...    318.0   

   original_language            original_title  \
0                NaN                       NaN   
7                 cn                      花樣年華   
19                en                    Vulgar   
22                en               Chicken Run   
25                en  The Million Dollar Hotel   

                                             overview  popularity  \
0                                                 NaN         NaN   
7   Two neighbors become intimate after discoverin...      23.460   
19  Vulgar is about a man who is a children's clow...       4.567   
22  The creators of Wallace & Gromit bring you an ...      38.196   
25  The Million Dollar Hotel starts with a jump fr...      11.318   

                                 production_companies  ... release_date  \
0                                                 NaN  ...          NaN   
7   [{'id': 539, 'logo_path': '/iPLtePguIzOPNtAWfT...  ...   2000-09-29   
19  [{'id': 102045, 'logo_path': None, 'name': 'Ch...  ...   2002-04-26   
22  [{'id': 521, 'logo_path': '/kP7t6RwGz2AvvTkvnI...  ...   2000-06-23   
25  [{'id': 1490, 'logo_path': None, 'name': 'Kint...  ...   2000-02-09   

        revenue  runtime                                   spoken_languages  \
0           NaN      NaN                                                NaN   
7    12854953.0     99.0  [{'english_name': 'Cantonese', 'iso_639_1': 'c...   
19      14904.0     87.0  [{'english_name': 'English', 'iso_639_1': 'en'...   
22  224834564.0     84.0  [{'english_name': 'English', 'iso_639_1': 'en'...   
25     105983.0    122.0  [{'english_name': 'English', 'iso_639_1': 'en'...   

      status                     title video  vote_average  vote_count  \
0        NaN                       NaN   NaN           NaN         NaN   
7   Released      In the Mood for Love   0.0         8.109      2163.0   
19  Released                    Vulgar   0.0         5.500        44.0   
22  Released               Chicken Run   0.0         6.700      4313.0   
25  Released  The Million Dollar Hotel   0.0         5.918       274.0   

    certification  
0             NaN  
7              PG  
19              R  
22              G  
25              R  

[5 rows x 21 columns]

## How many movies had at least some valid financial information 

In [51]:
# Select desired columns
selected_cols = ['title', 'revenue', 'budget']
result_df = filtered_df.loc[:, selected_cols]
result_df[:10]

title      revenue       budget
0                        NaN          NaN          NaN
7       In the Mood for Love   12854953.0     150000.0
19                    Vulgar      14904.0     120000.0
22               Chicken Run  224834564.0   45000000.0
25  The Million Dollar Hotel     105983.0    8000000.0
26    Mission: Impossible II  546388105.0  125000000.0
27                     X-Men  296339527.0   75000000.0
28                Titan A.E.   36754634.0   75000000.0
29  The Emperor's New Groove  169327687.0  100000000.0
33              Return to Me   36609995.0   24000000.0

In [52]:
df_filtered = df[(df['revenue'] > 0) & (df['budget'] > 0)]
# Group the filtered DataFrame by 'movie' and get the count of movies in each group
count_by_movie = df_filtered.groupby('title').size()

# Display the result
print(count_by_movie)

title
102 Dalmatians            1
15 Minutes                1
2001: A Space Travesty    1
28 Days                   1
3 Strikes                 1
                         ..
Xchange                   1
Y Tu Mamá También         1
Yajamana                  1
Yamakasi                  1
Zoolander                 1
Length: 402, dtype: int64


- There were 402 movies with at least one financial data. 

## How many movies are there in each of the certification categories (G/PG/PG-13/R)?

In [67]:
# counting the movie count for each certification category 
grouped_movies = filtered_df.groupby('certification')
count_movies = grouped_movies.size() 
print(count_movies) 
print(count_movies.sum())

certification
G         18
NR         8
PG        31
PG-13    117
R        144
dtype: int64
318


- There are 318 movies for each certification categories. 

## What is the average revenue per certification category?

In [71]:
# Set the display float format option to a non-scientific notation format
pd.set_option('display.float_format', lambda x: '%.2f' % x)

# Group the data by "certification" and calculate the average revenue for each group
avg_revenue_by_certification = filtered_df.groupby("certification")["revenue"].mean()
avg_revenue_by_certification

certification
G        97342643.33
NR       19369745.62
PG      125078003.97
PG-13   111073959.08
R        51735537.01
Name: revenue, dtype: float64

## What is the average budget per certification category?

In [72]:
# Set the display float format option to a non-scientific notation format
pd.set_option('display.float_format', lambda x: '%.2f' % x)

# Group the data by "certification" and calculate the average revenue for each group
avg_budget_by_certification = filtered_df.groupby("certification")["budget"].mean()
avg_budget_by_certification

certification
G       33777635.28
NR      11837750.00
PG      50362814.00
PG-13   46644422.11
R       26732737.15
Name: budget, dtype: float64